cpu/stm32l1/periph/i2c.c

```diff
@@ -142,91 +142,84 @@ static void _i2c_init(I2C_TypeDef *i2c, int ccr)
     i2c->CR1 |= I2C_CR1_PE;
 }
 
-static void _pin_config(GPIO_TypeDef *port, int pin_scl, int pin_sda)
+static void _pin_config(i2c_t dev)
 {
-    /* Set GPIOs to AF mode */
-    port->MODER &= ~(3 << (2 * pin_scl));
-    port->MODER |= (2 << (2 * pin_scl));
-    port->MODER &= ~(3 << (2 * pin_sda));
-    port->MODER |= (2 << (2 * pin_sda));
-
-    /* Set speed high*/
-    port->OSPEEDR |= (3 << (2 * pin_scl));
-    port->OSPEEDR |= (3 << (2 * pin_sda));
-
-    /* Set to push-pull configuration open drain*/
-    port->OTYPER |= (1 << pin_scl);
-    port->OTYPER |= (1 << pin_sda);
-
-    /* Enable pull-up resistors */
-    port->PUPDR &= ~(3 << (2 * pin_scl));
-    port->PUPDR |= (1 << (2 * pin_scl));
-    port->PUPDR &= ~(3 << (2 * pin_sda));
-    port->PUPDR |= (1 << (2 * pin_sda));
+    GPIO_TypeDef *port = i2c_config[dev].port;
+    int pin[2]; /**< 2 pins: scl, sda */
+    int af[2];  /**< 2 alternate functions: scl_af, sda_af */
+    int hl, i;
+
+    pin[0] = i2c_config[dev].scl;
+    pin[1] = i2c_config[dev].sda;
+    af[0] = i2c_config[dev].scl_af;
+    af[1] = i2c_config[dev].sda_af;
 
     /* Configure GPIOs to for the I2C alternate function */
-    if (pin_scl < 8) {
-        port->AFR[0] &= ~(0xf << (4 * pin_scl));
-        port->AFR[0] |= (I2C_0_SCL_AF << (4 * pin_scl));
-    }
-    else {
-        port->AFR[1] &= ~(0xf << (4 * (pin_scl - 8)));
-        port->AFR[1] |= (I2C_0_SCL_AF << (4 * (pin_scl - 8)));
-    }
+    for (i = 0; i < 2; i++) {
+        /* Set GPIOs to AF mode */
+        port->MODER &= ~(3 << (2 * pin[i]));
+        port->MODER |= (2 << (2 * pin[i]));
 
-    if (pin_sda < 8) {
-        port->AFR[0] &= ~(0xf << (4 * pin_sda));
-        port->AFR[0] |= (I2C_0_SDA_AF << (4 * pin_sda));
-    }
-    else {
-        port->AFR[1] &= ~(0xf << (4 * (pin_sda - 8)));
-        port->AFR[1] |= (I2C_0_SDA_AF << (4 * (pin_sda - 8)));
+        /* Set speed high*/
+        port->OSPEEDR |= (3 << (2 * pin[i]));
+
+        /* Set to push-pull configuration open drain*/
+        port->OTYPER |= (1 << pin[i]);
+
+        /* Enable pull-up resistors */
+        port->PUPDR &= ~(3 << (2 * pin[i]));
+        port->PUPDR |= (1 << (2 * pin[i]));
+
+        /* Set alternate function of the pin to I2C */
+        hl = (pin[i] < 8) ? 0 : 1;
+        port->AFR[hl] &= ~(0xf << ((pin[i] - (hl * 8)) * 4));
+        port->AFR[hl] |= (af[i] << ((pin[i] - (hl * 8)) * 4));
     }
 }
 
-static void _toggle_pins(GPIO_TypeDef *port, int pin_scl, int pin_sda)
+static void _toggle_pins(i2c_t dev)
```

Comment:
Necessity of this function discussed above. Can be removed if the above mentioned errata sheet doesn't affect this cpu